src/actions.cpp

```diff
@@ -108,14 +99,14 @@ bool Actions::registerEvent(Event* event, const pugi::xml_node& node)
 		uint16_t iterId = fromId;
 		uint16_t toId = pugi::cast<uint16_t>(toIdAttribute.value());
 
-		auto result = useItemMap.emplace(iterId, action);
+		auto result = useItemMap.emplace(iterId, *action);
```

Comment:
Why aren't you moving the action, like in line 86?

Reply:
Because in this part, its the fromid and toid code, I need to make a copy of the action for everytime it's inserted. Not ideal but I don't know how to do it better with a map.